WooCommerce/WooCommerceTests/System/WooConstantsTests.swift

```diff
@@ -12,7 +12,32 @@ final class WooConstantsTests: XCTestCase {
 
         // Then
         zip(allTrustedURLPaths, allTrustedURLs).forEach { (path, url) in
-            XCTAssertEqual(path.rawValue, url.absoluteString)
+            XCTAssertEqual(path.asURL().absoluteString, url.absoluteString)
         }
     }
+
+    func test_URLs_that_contains_special_characters_when_are_escaped_properly_then_do_not_return_nil() throws {
+        // Given
+        let urlsWithSpecialCharacters = [
+            WooConstants.URLs.inPersonPaymentsCashOnDeliveryFeedback,
+            WooConstants.URLs.inPersonPaymentsFirstTransactionFeedback,
+            WooConstants.URLs.inPersonPaymentsPowerUsersFeedback
```

Comment:
Nit: we should use test string instead of `WooConstants`, in case the URLs are changed. Also, we should use test URLs instead of actual URLs in case there are privacy issues.
```suggestion
            "https://test.com/woo-app-–-cod-survey",
```

Reply:
Definitely, changed here: https://github.com/woocommerce/woocommerce-ios/pull/8805/commits/0e4a799709ff01bbc48785e1adfbbfa8dd5f8bc2

Reply:
As an update, I had to add a testing string to `WooConstants.URL` in order to test these cases. The reason is because `asURL()` is a private extension, the one we get directly comes from AlamoFire and would crash if we pass a non-trusted URL. Our version seems was a re-implementation of this one.

```
▿ AFError
  ▿ invalidURL : 1 element
    - url : "https://test.com/test-–-survey"*/
``` 

Change on fc7dc937c68c36ab8bd1ca2f23e375adf8051c6e

Reply:
I didn't know `asURL()` is an internal method of `WooConstants` - in that case maybe it's better to add an extension method for `URL` and write tests for that extension instead.

Otherwise, you should add that testURL to an extension of `WooConstants` within the Test target - that looks better than adding it to the project @iamgabrielma.

Reply:
> I didn't know asURL() is an internal method of WooConstants - in that case maybe it's better to add an extension method for URL and write tests for that extension instead.

Sounds better indeed. Opened a new issue at https://github.com/woocommerce/woocommerce-ios/issues/8810 to deal with this separately